fetch.py

```diff
@@ -0,0 +1,152 @@
+import os
+import time
+import datetime
+import functools
+import zlib
+import requests
+import re
+from tqdm import tqdm
+"""
+Decorator for downloading or updating files required for a function.
+"""
+def reqfile(func=None, path=None, url=None):
+    if not func:
+        return functools.partial(fetch_factory, path=path, url=url)
+    @functools.wraps(test_func)
+    def wrapper(*args, **kwargs):
+        fetch(path, url)
+        return func(*args, **kwargs)
+    return wrapper
+
+"""
+Downloads a file from url to a given path.
+Also shows a progress bar to keep track.
+"""
+def downfile(path, url):
+    r = requests.get(url, stream=True)
+    r.raise_for_status()
+    total_size = int(r.headers.get('content-length', 0))
+    with open(path, 'wb') as f:
+        chunk_size = 64*1024
+        pbar = tqdm(total=total_size, unit='B', unit_scale=True)
+        for data in r.iter_content(chunk_size):
+            f.write(data)
+            pbar.update(chunk_size)
+
+"""
+Checks a given path for file and downloads if a url is given or file name
+is an accession id. Also downloads the file if the remote location has a
+more recent version. Gives an IOError if path does not exist and an accession id
+can not be found.
+    Usage:
+        fetch('NC_000853.txt')
+        fetch('DRDatabase.txt', 'http://crispr.i2bc.paris-saclay.fr/crispr/BLAST/DR/DRdatabase')
```

Comment:
Do you run this in python or from the command line? Can you add a bit more detail to the usage part here?